modules/services/compton.nix

```diff
@@ -1,43 +0,0 @@
-{ config, lib, pkgs, ... }:
-
-with lib; {
-  imports = let
-    old = n: [ "services" "compton" n ];
-    new = n: [ "services" "picom" n ];
-  in [
-    (mkRenamedOptionModule (old "activeOpacity") (new "activeOpacity"))
-    (mkRenamedOptionModule (old "backend") (new "backend"))
-    (mkRenamedOptionModule (old "blur") (new "blur"))
-    (mkRenamedOptionModule (old "blurExclude") (new "blurExclude"))
-    (mkRenamedOptionModule (old "extraOptions") (new "extraOptions"))
-    (mkRenamedOptionModule (old "fade") (new "fade"))
-    (mkRenamedOptionModule (old "fadeDelta") (new "fadeDelta"))
-    (mkRenamedOptionModule (old "fadeExclude") (new "fadeExclude"))
-    (mkRenamedOptionModule (old "fadeSteps") (new "fadeSteps"))
-    (mkRenamedOptionModule (old "inactiveDim") (new "inactiveDim"))
-    (mkRenamedOptionModule (old "inactiveOpacity") (new "inactiveOpacity"))
-    (mkRenamedOptionModule (old "menuOpacity") (new "menuOpacity"))
-    (mkRenamedOptionModule (old "noDNDShadow") (new "noDNDShadow"))
-    (mkRenamedOptionModule (old "noDockShadow") (new "noDockShadow"))
-    (mkRenamedOptionModule (old "opacityRule") (new "opacityRule"))
-    (mkRenamedOptionModule (old "package") (new "package"))
-    (mkRenamedOptionModule (old "refreshRate") (new "refreshRate"))
-    (mkRenamedOptionModule (old "shadow") (new "shadow"))
-    (mkRenamedOptionModule (old "shadowExclude") (new "shadowExclude"))
-    (mkRenamedOptionModule (old "shadowOffsets") (new "shadowOffsets"))
-    (mkRenamedOptionModule (old "shadowOpacity") (new "shadowOpacity"))
-    (mkChangedOptionModule (old "vSync") (new "vSync") (v: v != "none"))
-  ];
-
-  options.services.compton.enable = mkEnableOption "Compton X11 compositor" // {
-    visible = false;
-  };
-
-  config = mkIf config.services.compton.enable {
```

Comment:
IMO we should gatekeep the module behind a stateVersion check. If stateVersion >= 22.05 then error else warn.

Reply:
I am not sure how this would work without making this module compatible with the `config.services.picom` itself (because this is a breaking change). So it would break either way.

Also, this module is deprecated since 2020-03-24 according to the news entry. Everybody that used this module should already have migrated.

Reply:
If you want I can keep this module but make it throw if activate. This should give a better error than just deleting the file.

But I am not sure if it is worth it, since as I said, this module is deprecated for a long time already.